### PR TITLE
GameMode

### PR DESCRIPTION
**Pull request by onesounds**
https://user-images.githubusercontent.com/6903107/142577704-384ffac7-376d-4b52-8e36-bd75ec0afad5.mp4

In my case, I tend to turn off all tray programs that are likely to affect game play.
Since games are often played without using the entire screen, this setting is necessary.

- Add "Game Mode" in Tray Menu (Toggle)
- The flow-on/custom query hotkey shortcut does not work when the game mode is on.
- it will change Tray Icon to red when turn on Game Mode

**ETC**
- "Pause Hotkey" is better name?
- It would be nice if we could pause the some part (indexing?) that can affect the speed.

### CODE REVIEW

File: Flow.Launcher/ViewModel/MainViewModel.cs [hunk 1]
```diff
@@ -44,6 +44,7 @@ public class MainViewModel : BaseModel, ISavable
         private readonly History _history;
         private readonly UserSelectedRecord _userSelectedRecord;
         private readonly TopMostRecord _topMostRecord;
+        private MainWindow _mainWindow;
```
Comment (jjw24) [inline Flow.Launcher/ViewModel/MainViewModel.cs:47]: i dont think this is used right, it's private and not referenced in this vm
```suggestion
```

File: Flow.Launcher/ViewModel/MainViewModel.cs [hunk 2]
```diff
@@ -304,6 +305,8 @@ private void InitializeKeyCommands()
         public ResultsViewModel ContextMenu { get; private set; }
         public ResultsViewModel History { get; private set; }
 
+        public bool gameModeStatus = false;
```
Comment (jjw24) [inline Flow.Launcher/ViewModel/MainViewModel.cs:308]: tip, bools are initialised as false, so you dont need to set it to false.
we can use a getter and setter here actually
```suggestion
        public bool GameModeStatus { get; set; }
```

Comment (jjw24) [reply Flow.Launcher/ViewModel/MainViewModel.cs:308]: public variables use pascal case where as private variables are lower camel case.

File: Flow.Launcher/Properties/Resources.resx [hunk 3]
```diff
@@ -112,13 +112,16 @@
     <value>2.0</value>
   </resheader>
   <resheader name="reader">
-    <value>System.Resources.ResXResourceReader, System.Windows.Forms, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+    <value>System.Resources.ResXResourceReader, System.Windows.Forms, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
```
Comment (jjw24) [inline Flow.Launcher/Properties/Resources.resx:115]: these version changes are intentional?

File: Flow.Launcher/Flow.Launcher.csproj [hunk 4]
```diff
@@ -104,6 +104,9 @@
     <ProjectReference Include="..\Flow.Launcher.Plugin\Flow.Launcher.Plugin.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <Resource Include="Images\gamemode.ico" />
```
Comment (jjw24) [inline Flow.Launcher/Flow.Launcher.csproj:108]: we dont need to do this separately because up top it's already including .ico files :)

**Comment by jjw24** [comment]
i will make the recommendations via direct commits if it's ok.

File: Flow.Launcher/Properties/Resources.resx (hunk 3 above)
Comment (onesounds) [reply Flow.Launcher/Properties/Resources.resx:115]: no. visual studio did it.

**Comment by onesounds** [comment]
> i will make the recommendations via direct commits if it's ok.

yeah, please.

**Comment by jjw24** [comment]
not related to this pr but i noticed ignore hotkey when in full screen mode is not working, neither in dev. i will fix that up later

**Review by jjw24 — APPROVED** [review]

**Pull request merged**